lisp/mastodon-tl.el

```diff
@@ -281,39 +307,54 @@ Move forward (down) the timeline unless BACKWARD is non-nil."
 (defun mastodon-tl--more ()
   "Append older toots to timeline."
   (interactive)
-  (let* ((point-before (point))
-         (tl (mastodon-tl--timeline-name))
-         (id (mastodon-tl--oldest-id))
-         (json (mastodon-tl--more-json tl id)))
-    (when json
-      (with-current-buffer (current-buffer)
+  (with-current-buffer (current-buffer)
+    (let* ((point-before (point))
+           (endpoint (mastodon-tl--get-endpoint))
+           (update-function (mastodon-tl--get-update-function))
+           (id (mastodon-tl--oldest-id))
+           (json (mastodon-tl--more-json endpoint id)))
+      (when json
         (let ((inhibit-read-only t))
           (goto-char (point-max))
-          (mastodon-tl--timeline json)
+          (funcall update-function json)
           (goto-char point-before)
           (mastodon-tl--goto-next-toot))))))
 
 (defun mastodon-tl--update ()
   "Update timeline with new toots."
   (interactive)
-  (let* ((tl (mastodon-tl--timeline-name))
-         (id (mastodon-tl--newest-id))
-         (json (mastodon-tl--updated-json tl id)))
-    (when json
-      (with-current-buffer (current-buffer)
+  (with-current-buffer (current-buffer)
+    (let* ((endpoint (mastodon-tl--get-endpoint))
+           (update-function (mastodon-tl--get-update-function))
+           (id (mastodon-tl--newest-id))
+           (json (mastodon-tl--updated-json endpoint id)))
+      (when json
         (let ((inhibit-read-only t))
           (goto-char (point-min))
-          (mastodon-tl--timeline json))))))
+          (funcall update-function json))))))
+
+(defun mastodon-tl--init(buffer-name endpoint update-function)
+  "Initialize BUFFER-NAME with timeline targeted by ENDPOINT.
 
-(defun mastodon-tl--get (timeline)
+UPDATE-FUNCTION is used to recive more toots."
+  (let ((buffer (mastodon-tl--get buffer-name endpoint update-function)))
+    (with-current-buffer buffer
+      (make-local-variable 'mastodon-buffer-spec)
+      (setq mastodon-buffer-spec
+            `(buffer-name ,buffer-name
+                          endpoint ,endpoint update-function
+                          ,update-function)))))
+
+(defun mastodon-tl--get (buffer-name endpoint update-function)
```

Comment:
I think I am going to envelop `mastodon-tl--get` into `mastodon-tl--init` since it has no purpose on its own.